libbroker/broker/publisher.cc

```diff
@@ -209,48 +86,38 @@ size_t publisher::free_capacity() const {
 }
 
 detail::native_socket publisher::fd() const {
-  return dptr(queue_)->fd();
+  return impl_->write_fd();
 }
 
 void publisher::drop_all_on_destruction() {
-  drop_on_destruction_ = true;
+  // nop
 }
 
-void publisher::publish(const data& x) {
-  auto msg = make_data_message(topic_, x);
-  log::endpoint::debug("publish", "publishing {}", msg);
-  dptr(queue_)->push(caf::make_span(&msg, 1));
+void publisher::publish(const data& val) {
+  impl_->publish(make_data_message(dst_, val));
 }
 
-void publisher::publish(const std::vector<data>& xs) {
-  std::vector<data_message> msgs;
-  msgs.reserve(xs.size());
-  for (auto& x : xs)
-    msgs.push_back(make_data_message(topic_, x));
-  log::endpoint::debug("publish-batch", "publishing a batch of size {}",
-                       xs.size());
-  dptr(queue_)->push(msgs);
+void publisher::publish(const std::vector<data>& vals) {
+  for (auto& val : vals) {
+    impl_->publish(make_data_message(dst_, val));
+  }
 }
 
-void publisher::publish(set_builder&& x) {
-  auto msg = std::move(x).build_envelope(topic_.string());
-  dptr(queue_)->push(caf::make_span(&msg, 1));
+void publisher::publish(set_builder&& content) {
+  impl_->publish(std::move(content).build_envelope(dst_.string()));
 }
 
-void publisher::publish(table_builder&& x) {
-  auto msg = std::move(x).build_envelope(topic_.string());
-  dptr(queue_)->push(caf::make_span(&msg, 1));
+void publisher::publish(table_builder&& content) {
+  impl_->publish(std::move(content).build_envelope(dst_.string()));
 }
 
-void publisher::publish(list_builder&& x) {
-  auto msg = std::move(x).build_envelope(topic_.string());
-  dptr(queue_)->push(caf::make_span(&msg, 1));
+void publisher::publish(list_builder&& content) {
+  impl_->publish(std::move(content).build_envelope(dst_.string()));
 }
 
 void publisher::reset() {
-  if (queue_) {
-    dptr(queue_)->buf().close();
-    queue_.reset();
+  if (impl_ == nullptr) {
```

Comment:
```suggestion
  if (impl_ != nullptr) {
```

This doesn't seem to make a difference, but spotted while skimming.